File: packages/firebase-messaging/platforms/ios/src/TNSFirebaseMessaging.m
#import "TNSFirebaseMessaging.h"
@implementation TNSFirebaseMessaging

BOOL ShowNotificationsWhenInForeground = NO;
BoolCallback RegisterDeviceForRemoteMessagesCallback = nil;
DictionaryCallback OnNotificationTapCallback = nil;
DictionaryCallback OnMessageCallback = nil;
StringCallback OnTokenCallback = nil;

+ (void)setShowNotificationsWhenInForeground:(BOOL)showNotificationsWhenInForeground {
    ShowNotificationsWhenInForeground = showNotificationsWhenInForeground;
}

+ (BOOL)showNotificationsWhenInForeground {
    return ShowNotificationsWhenInForeground;
}

+ (BoolCallback)registerDeviceForRemoteMessagesCallback {
    return RegisterDeviceForRemoteMessagesCallback;
}

+ (void)setRegisterDeviceForRemoteMessagesCallback:(BoolCallback)callback {
    RegisterDeviceForRemoteMessagesCallback = callback;
}

+ (DictionaryCallback)onNotificationTapCallback {
    return OnNotificationTapCallback;
}

+ (void)setOnNotificationTapCallback:(DictionaryCallback)callback {
    OnNotificationTapCallback = callback;
}

+ (DictionaryCallback)onMessageCallback {
    return OnMessageCallback;
}

+ (void)setOnMessageCallback:(DictionaryCallback)callback {
    OnMessageCallback = callback;
}

+ (StringCallback)onTokenCallback {
    return OnTokenCallback;
}

+ (void)setOnTokenCallback:(StringCallback)callback {
    OnTokenCallback = callback;
}


+ (NSString *)APNSTokenToString:(NSData *)tokenData {
    const char *data = [tokenData bytes];

    NSMutableString *token = [NSMutableString string];
    for (NSInteger i = 0; i < tokenData.length; i++) {
      [token appendFormat:@"%02.2hhX", data[i]];
    }

    return token;
}
@end

File: packages/firebase-messaging/platforms/ios/src/TNSFirebaseMessaging.swift
import Foundation
import UIKit


@objc(TNSFirebaseMessaging)
@objcMembers
public class TNSFirebaseMessaging: NSObject {
    public static var showNotificationsWhenInForeground = false
    
    @objc public static var registerDeviceForRemoteMessagesCallback: ((Bool, NSError?) -> Void)? = nil
    
    @objc public static var onNotificationTapCallback: ((NSDictionary) -> Void)? = nil
    
    @objc public static var onMessageCallback: ((NSDictionary) -> Void)? = nil
    
    public static var onTokenCallback: ((String) -> Void)? = nil
    
    
    @objc public static func APNSTokenToString(_ tokenData: NSData)-> String {
        var token = ""
        for tokenByte in tokenData {
            token.append(contentsOf:  String(format: "%02.2hhX", tokenByte))
        }
        return token
    }
}
